packages/mobile/App/services/sync/utils/buildFromSyncRecord.ts

```diff
@@ -29,28 +29,25 @@ const mapFields = (mapping: [string, string][], obj: { [key: string]: unknown })
   return newObj;
 };
 
-export const buildFromSyncRecord = (
-  model: typeof BaseModel,
+export const pickBySelectedColumns = (
   { data }: SyncRecord,
+  includedColumns: string[],
 ): DataToPersist => {
-  // find columns to include
-  const includedColumns = extractIncludedColumns(model);
-  // populate `fieldMapping` with `RelationId` to `Relation` mappings
-  // (not necessary for `IdRelation`)
-  const fieldMapping = getRelationIdsFieldMapping(model);
-
-  const dbRecord = mapFields(fieldMapping, pick(data, includedColumns));
-  return dbRecord;
+  const record = pick(data, includedColumns);
+  return record;
 };
 
-export const buildForRawInsertFromSyncRecord = (
+export const buildFromSyncRecords = (
   model: typeof BaseModel,
-  { data, isDeleted }: SyncRecord,
-): DataToPersist => {
+  records: SyncRecord[],
+): DataToPersist[] => {
   const includedColumns = extractIncludedColumns(model);
-  const record = pick(data, includedColumns);
-  record.isDeleted = isDeleted;
-  return record;
+  // populate `fieldMapping` with `RelationId` to `Relation` mappings
+  // (not necessary for `IdRelation`)
+  const fieldMapping = getRelationIdsFieldMapping(model);
+  return records.map(record =>
+    mapFields(fieldMapping, pickBySelectedColumns(record, includedColumns)),
```

Comment:
I know I pushed in other places for pulling stuff out where maps get too dense, but in this case I think it's _more_ reading effort to understand `pickBySelectedColumns` than just reading the `pick` directly
```suggestion
    mapFields(fieldMapping, pick(record.data, includedColumns)),
```